### PR TITLE
Feature/cmn 293 translate loan list plus cleanup

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

Link til låneliste ? -> https://platform.dandigbib.org/issues/5387

#### Description

- Givet nogle strenge mere meningsfulde navne
- Rettede stavefejl (titel -> title) 
- Omdøbt paginate til paginatedLoans as this is more correct
- fixet en aria-described-by
- flyttet en described-by så den ikke bliver gentaget for hvert materiale (hvilket er skidt med id'er), men i stedet er et "centralt" sted
- bruger interpolation hurra
- erstat refetch med invalideret cache
- add isDigital and getStatusText

### CODE REVIEW

File: src/apps/reservation-list/reservation-material/reservation-status.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ const ReservationStatus: FC<ReservationStatusProps> = ({
         className="list-reservation__counter"
         // todo create meaningful aria-explanation
       >
-        <StatusCircleIcon color={color} percent={percent}>
+        <StatusCircleIcon ariaLabel="todo" color={color} percent={percent}>
```
Comment (sinejespersen) [inline src/apps/reservation-list/reservation-material/reservation-status.tsx:27]: Already fixed in another branch

**Review by jeppekroghitk — APPROVED** [review]

File: src/apps/loan-list/list/list.tsx [hunk 2]
```diff
@@ -70,7 +68,7 @@ const List: FC<ListProps> = ({
 
   return (
     <>
-      <div className="dpl-list-buttons m-32">
+      <div className="dpl-list-buttons my-32">
```
Comment (spaceo) [inline src/apps/loan-list/list/list.tsx:71]: What is `my-*`?

File: src/apps/loan-list/materials/selectable-material.tsx [hunk 3]
```diff
@@ -67,27 +67,26 @@ const SelectableMaterial: FC<SelectableMaterialProps & MaterialProps> = ({
...
-            </span>
+            </>
           )}
           {dueDate && (
             <StatusBadge
               dueDate={dueDate}
-              neutralText={`${t("loanListToBeDeliveredMaterialText")} 
-            ${formatDate(dueDate)}`}
+              neutralText={t("groupModalDueDateMaterialText", {
+                placeholders: { "@count": formatDate(dueDate) }
```
Comment (spaceo) [inline src/apps/loan-list/materials/selectable-material.tsx:88]: `@count`pointing at a date? Is it on purpose or is it a mistake?

File: src/apps/loan-list/materials/stackable-material/material-status.tsx [hunk 4]
```diff
@@ -32,7 +27,13 @@ const MaterialStatus: FC<MaterialStatusProps> = ({
...
           />
           <p className="text-small-caption" id="due-date">
-            {dueDateLabel} {formatDate(dueDate)}
+            {isDigital(loan)
+              ? t("loanListToBeDeliveredDigitalMaterialText", {
+                  placeholders: { "@count": formatDate(dueDate) }
+                })
+              : t("loanListToBeDeliveredText", {
+                  placeholders: { "@count": formatDate(dueDate) }
+                })}
```
Comment (spaceo) [inline src/apps/loan-list/materials/stackable-material/material-status.tsx:36]: You could also make a function to get the text, it is almost similar.
But you are also pointing `@count`to a date here. I really do not get it :)

File: src/apps/loan-list/materials/utils/status-circle.tsx [hunk 5]
```diff
@@ -27,13 +27,29 @@ const StatusCircle: FC<StatusCircleProps> = ({ loanDate, dueDate }) => {
...
+    daysBetweenTodayAndDue > 0 ? daysBetweenTodayAndDue : 0;
   return (
-    <StatusCircleIcon percent={percent} color={color as string}>
+    <StatusCircleIcon
+      ariaLabel={t("loanListStatusCircleAriaLabelText", {
+        count: daysUntilDuedate,
+        placeholders: { "@count": daysUntilDuedate }
+      })}
+      percent={percent}
+      color={color as string}
```
Comment (spaceo) [inline src/apps/loan-list/materials/utils/status-circle.tsx:39]: Is it possible to assure that color is a string earlier on?

File: src/apps/loan-list/materials/utils/status-circle.tsx [hunk 6]
```diff
@@ -27,13 +27,29 @@ const StatusCircle: FC<StatusCircleProps> = ({ loanDate, dueDate }) => {
...
+    >
       <span className="counter__value">
-        {daysBetweenTodayAndDue > 0 ? daysBetweenTodayAndDue : 0}
+        {/* I am not using string interpolation here because of styling */}
+        {/* if somehow it is possible to break text in one div into two lines */}
+        {/* where the first line has another font size AND is only the first "word" */}
+        {/* then this should be changed to do that */}
```
Comment (spaceo) [inline src/apps/loan-list/materials/utils/status-circle.tsx:45]: 👍 For commenting this

File: src/apps/loan-list/materials/utils/warning-bar.tsx [hunk 7]
```diff
@@ -2,25 +2,27 @@ import React, { FC } from "react";
...
   linkText: string;
   overdueText: string;
 }
 
 const WarningBar: FC<WarningBarProps> = ({ linkText, overdueText }) => {
-  /* todo link til gebyrer */
-  const url = new URL("/", getCurrentLocation());
-
+  const { feesPageUrl } = useUrls();
+  const feesPage = new URL(feesPageUrl, getCurrentLocation());
```
Comment (spaceo) [inline src/apps/loan-list/materials/utils/warning-bar.tsx:14]: useUrl should deliver URL objects. Are you shure it is necessary to create your own object as well?

File: src/apps/loan-list/modal/due-date-loans-modal.tsx [hunk 8]
```diff
@@ -42,25 +44,23 @@ const DueDateLoansModal: FC<DueDateLoansModalProps> = ({
               </div>
               <div>
                 <h1 className="modal-loan__title text-header-h2">
-                  {t("dueDateRenewLoanModalHeaderText")} {formatDate(dueDate)}
+                  {t("groupModalDueDateHeaderText", {
+                    placeholders: { "@date": formatDate(dueDate) }
```
Comment (spaceo) [inline src/apps/loan-list/modal/due-date-loans-modal.tsx:48]: Yay this I get :)

File: src/apps/loan-list/modal/renew-loans-modal-content.tsx [hunk 9]
```diff
@@ -53,18 +49,18 @@ const RenewLoansModalContent: FC<RenewLoansModalContentProps> = ({
       {
         onSuccess: (result) => {
           if (result) {
+            queryClient.invalidateQueries(getGetLoansV2QueryKey());
```
Comment (spaceo) [inline src/apps/loan-list/modal/renew-loans-modal-content.tsx:52]: 👍

File: src/apps/loan-list/utils/helpers.ts [hunk 10]
```diff
@@ -63,6 +75,11 @@ export const queryMatchesFaust = (query: string | null) => {
   return returnValue;
 };
 
+export const isDigital = (loan: LoanType) => {
+  if (loan.identifier) return true;
+  return false;
+};
+
```
Comment (spaceo) [inline src/apps/loan-list/utils/helpers.ts:82]:
```suggestion
export const isDigital = (loan: LoanType) => Boolean(loan.identifier);
```

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks good. Have some questions though.

File: src/apps/loan-list/materials/selectable-material.tsx (hunk 3 above)
Comment (sinejespersen) [reply src/apps/loan-list/materials/selectable-material.tsx:88]: mistake, ill fix

File: src/apps/loan-list/materials/utils/warning-bar.tsx (hunk 7 above)
Comment (sinejespersen) [reply src/apps/loan-list/materials/utils/warning-bar.tsx:14]: ill fix

File: src/apps/loan-list/list/list.tsx (hunk 2 above)
Comment (sinejespersen) [reply src/apps/loan-list/list/list.tsx:71]: https://github.com/danskernesdigitalebibliotek/dpl-design-system/blob/58e4403db616cfb6c09213951a78e0a75ef44c72/src/stories/Library/spacing/spacing.scss#L57

**Review by spaceo — APPROVED** [review]

**Pull request merged**